packages/vaex-core/vaex/column.py

```diff
@@ -0,0 +1,252 @@
+        if stop <= offset + len(current_df):
+            if current_df.filtered:  # TODO this may get slow! we're evaluating everything
+                warnings.warn("might be slow, you have concatenated dfs with a filter set")
+            return current_df.evaluate(self.column_name, i1=start - offset, i2=stop - offset)
+        else:
+            if self.is_masked:
+                copy = np.ma.empty(stop - start, dtype=dtype)
+                copy.fill_value = self.fill_value
+            else:
+                copy = np.zeros(stop - start, dtype=dtype)
+            copy_offset = 0
+            # print("!!>", start, stop, offset, len(current_df), current_df.columns[self.column_name])
+            while offset < stop:  # > offset + len(current_df):
+                # print(offset, stop)
+                if current_df.filtered:  # TODO this may get slow! we're evaluating everything
+                    warnings.warn("might be slow, you have concatenated DataFrames with a filter set")
+                part = current_df.evaluate(self.column_name, i1=start-offset, i2=min(len(current_df), stop - offset))
+                # print "part", part, copy_offset,copy_offset+len(part)
+                copy[copy_offset:copy_offset + len(part)] = part
+                # print copy[copy_offset:copy_offset+len(part)]
+                offset += len(current_df)
+                copy_offset += len(part)
+                start = offset
+                if offset < stop:
+                    current_df = next(dfs)
+            return copy
+
+try:
+    str_type = unicode
+except:
+    str_type = str
+
+use_c_api = True
+
+class ColumnString(Column):
+    pass
+
+class ColumnStringArray(Column):
+    """Wraps a numpy array with dtype=object, containing all strings"""
+    def __init__(self, array):
+        # assert vaex.strings.all_strings(array, allow)
+        self.array = array
+        self.string_sequence = vaex.strings.StringArray(self.array)
+
+    def __getitem__(self, slice):
+        return ColumnString(self.array[slice])
+
+    def to_numpy(self):
+        return self.array
+
+def _to_string(string_bytes):
+    if six.PY2:
+        return unicode(buffer(string_bytes))
+    else:
+        return bytes(memoryview(string_bytes)).decode('utf8')
+
+class ColumnStringArrow(ColumnString):
+    """Column that unpacks the arrow string column on the fly"""
+    def __init__(self, indices, bytes, length=None, offset=0):
+        self.indices = indices
+        self.offset = offset  # to avoid memory copies in trim
+        self.bytes = bytes
+        self.dtype = str_type
+        self.length = length if length is not None else len(indices) - 1
+        self.shape = (self.__len__(),)
+        self.nbytes = self.bytes.nbytes + self.indices.nbytes
+        self.string_sequence = vaex.strings.StringList(self.bytes, self.indices, self.length, self.offset)
+
+    def __len__(self):
+        return self.length
+
+    def __getitem__(self, slice):
```

Comment:
This looks familiar to https://github.com/xhochy/fletcher/blob/cbc35c1b168ad140bcaabdd8efd22a2425b52a03/fletcher/base.py#L331 Not the exact same code but I guess a bit of this shoudl go upstream

Reply:
Ah, thanks for the pointer. Yes, that would be great. With upstream you mean in arrow?
I have the issue of having to attack a bit of a broader problem with strings than arrow (other formats, such as numpy fixed strings), so it goes a bit beyond arrow, but I hope in the end I can contribute something back!

Reply:
Yes, upstream is Arrow. `fletcher` is my toy project to better understand what users like Pandas or Vaex could need from Arrow.